frame/contracts/src/lib.rs

```diff
@@ -629,8 +638,13 @@ decl_module! {
 			};
 
 			// If poking the contract has lead to eviction of the contract, give out the rewards.
-			if Rent::<T>::snitch_contract_should_be_evicted(&dest, handicap)? {
-				T::Currency::deposit_into_existing(&rewarded, T::SurchargeReward::get()).map(|_| ())
+			if let Some(rent_payed) = Rent::<T>::try_eviction(&dest, handicap)? {
+				T::Currency::deposit_into_existing(
+					&rewarded,
+					T::SurchargeReward::get().min(rent_payed),
```

Comment:
here it is not rent_payed but rather `rent_payed + rent_to_pay`.

isn't it an issue if like a contract never payed its rent and the claim_surchage ppl are rather slow.

On the condition claim_surchage ppl are very slow to claim then it can happen that surchage reward is more than cost of contract.

If the hypothesis "claim_surchage ppl are very slow" is very unlikely, then all fine, but otherwise maybe we can just cap with the actual rent_payed instead of total_rent.

Reply:
Your conclusion is correct but your premise is wrong (see previous comment).

Reply:
indeed